### PR TITLE
updated message_id and added message_key

### PR DESCRIPTION
**Pull request by paulhandy**

### CODE REVIEW

**Comment by paulhandy** [comment]
Added bindings

File: mam/src/mam.rs [hunk 1]
```diff
@@ -7,18 +7,31 @@ use auth::*;
...
+pub fn message_id<T, C>(key: &mut [T], curl: &mut C)
 where
     T: Copy + Clone + Sized,
     C: Curl<T>,
 {
-    for key in keys {
-        curl.absorb(key.as_slice());
-    }
-    let mask = curl.rate().to_vec();
+    curl.absorb(key);
```
Comment (th0br0) [inline mam/src/mam.rs]: please assert that key is at least of HASH_LENGTH length

Comment (th0br0) [inline mam/src/mam.rs]: also, not sure whether we want to overwrite the merkle root (which this does!?)

File: mam/src/mam.rs [hunk 2]
```diff
@@ -7,18 +7,31 @@ use auth::*;
 use mask::*;
 use errors::*;
 
-pub fn message_id<T, C>(keys: &[Vec<T>], curl: &mut C) -> Vec<T>
+pub fn message_key(key: &[Trit], out: &mut [Trit], index: usize) -> Result<usize, MamError> {
```
Comment (th0br0) [inline mam/src/mam.rs]: add comment

**Pull request closed without merging**